axelrod/actions.py

```diff
@@ -9,31 +9,65 @@
 C, D = Actions.C, Actions.D
 """
 
-# Type alias for actions.
-Action = str
+from enum import Enum
+from typing import Iterable
+
+
+class UnknownActionError(ValueError):
+    def __init__(self, *args):
+        super(UnknownActionError, self).__init__(*args)
+
+
+class Actions(Enum):
+
+    C = 1
+    D = 0
+
+    def __bool__(self):
+        return bool(self.value)
 
+    def __repr__(self):
+        return '{}'.format(self.name)
 
-class Actions(object):
-    C = 'C'  # type: Action
-    D = 'D'  # type: Action
+    def __str__(self):
+        return '{}'.format(self.name)
+
+    def flip(self):
+        """Returns the opposite Action. """
```

Comment:
I believe i originally wrote this as also using a dict. Is it better to use the if statements or the dict?  I changed it to follow the style of the function below.

```
    def flip(self):
        """Returns the opposite Action. """
        opposites = {Actions.C: Actions.D,
                     Actions.D: Actions.C}
        return opposites[self]
```

should i use this code instead?